### ISSUE TITLE
Subfolders not working

### ISSUE THREAD

**Issue by ghost** [grrttedwards/wow-addon-updater#34]
The example in the readme with Tukui ElvUI is not working with the current master version, it cant find the subfolder.

**Comment by Hjaltesorgenfrei** [comment]
I created a possible fix [here](https://github.com/Hjaltesorgenfrei/wow-addon-updater/blob/subfolder-fix/updater/manager/addon_manager.py) 
```py
  def extract_to_addons(self, zipped: zipfile.ZipFile, subfolder, foldername):
        if subfolder:
            subfolder = subfolder[0]
            destination_dir = join(self.WOW_ADDON_LOCATION, subfolder)
            folder = self.WOW_ADDON_LOCATION + '/' + foldername + '/'
            if exists(folder):
                rmtree(folder)
            print(folder)
            for file in zipped.namelist():
                if file.startswith(subfolder):
                    name = folder + file.replace(subfolder, '')
                    if name[-1:] is '/':
                        if not exists(name):
                            makedirs(name)
                            continue
                    member = zipped.open(file)
                    with open(folder + file.replace(subfolder, ''), 'wb') as outfile:
                        # copy it directly to the output directory,
                        # without creating the intermediate directory
                        copyfileobj(member, outfile)
        else:
            zipped.extractall(self.WOW_ADDON_LOCATION)
```

**Comment by Hjaltesorgenfrei** [comment]
I should add that this makes the program work with addons like this one https://github.com/smp4903/FiveSecondRule
As long as the name of the Repo is what the addon folder should be called.

**Comment by Ghosthree3** [comment]
It would be nice if we could get multi-subfolder support too, so instead of:
https://git.tukui.org/elvui/elvui|ElvUI
https://git.tukui.org/elvui/elvui|ElvUI_OptionsUI
We could just have:
https://git.tukui.org/elvui/elvui|ElvUI|ElvUI_OptionsUI
This way we avoid downloading the archive twice for no reason. Granted, multi-threading has made this less of an issue.

**Comment by Hjaltesorgenfrei** [comment]
I will look into it.

**Comment by grrttedwards** [comment]
I think I might like to change that delimiter (or at least add an alternative) as a single space ` ` character. It makes the URLs in your list a bit more readable than the super thin `|` pipe. Multi-subfolder would also be cool.

**Comment by Hjaltesorgenfrei** [comment]
My current [subfolder branch](https://github.com/Hjaltesorgenfrei/wow-addon-updater/tree/subfolder-fix) works with Multi-subfolder. You can test it with: 
https://git.tukui.org/elvui/elvui|elvui-master/ElvUI|elvui-master/ElvUI_OptionsUI

But i'm not really happy with how it handles deleting the old folder, currently i use shutil.rmtree() which i think is unsafe.

**Comment by grrttedwards** [comment]
That's a good base. I'll take that and run with it. Looks like there are a few ways to do that.

Also, this raises a point I didn't notice... zip.extract does not wipe the destination directory before performing the copy, which means it won't always install to a clean directory. Addons can leave old files version-to-version. I'll make an issue for that.

**Comment by Hjaltesorgenfrei** [comment]
Checking that the path that gets deleted is in the chosen directory for addons is probaly a good idea. We could do this by getting the os.path.realpath on the addon path and making sure that its starts with addon location.
Just to avoid deleting folders outside the addon directory.

**Comment by grrttedwards** [comment]
Fixed by #42 with some tests
